clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -87,6 +87,21 @@
 
     public static final long NO_IDLE_TIMEOUT_MS = -1;
 
+    private enum CloseMode {
+        LOCAL_CLOSE(false, false),
```

Comment:
Since only two booleans affect decision making, I think reducing the number of enum to 4 would be more readable.

Reply:
Yeah, I agree. Here's my attempt:

FAILED_SEND, LOCAL_CLOSE -> FORCED
EXPIRED, DISCONNECTED -> GRACEFUL
DISCONNECTED_AND_FAILED_SEND -> NOTIFY_ONLY

If we can't come up with good names, we could also revert and use the two boolean parameters.

Reply:
It would be difficult to give meaningful enum names in short time.

Using two boolean parameters seems to be better for the moment.

If there is ever need for 3rd boolean, we can refactor the code.

Reply:
@hachikuji @tedyu @ijuma OK, I am going to revert and commit for now. We could improve on it later if we make updates to the code again.

Reply:
I think the enum approach was better. I'd go with `DISCARD`, `GRACEFUL` and `NOTIFY_ONLY`.

Reply:
@ijuma Thanks for the review. I will submit another PR with the update to use enum. But since this is a cosmetic change, I think it doesn't have to block any of the releases. Perhaps `DISCARD_STAGED_RECEIVES`, `GRACEFUL` and `NOTIFY_ONLY`?

Reply:
Yeah, no rush.